File: src/main/java/ch/loewenfels/issuetrackingsync/controller/StatisticsController.java
package ch.loewenfels.issuetrackingsync.controller;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class StatisticsController {
    @GetMapping("/statistics")
    public Map<String, Serializable> getStatistics() {
        final Map<String, Serializable> result = new HashMap<>();
        return result;
    }
}

File: src/main/kotlin/ch/loewenfels/issuetrackingsync/controller/StatisticsController.kt
package ch.loewenfels.issuetrackingsync.controller

import ch.loewenfels.issuetrackingsync.INTERNAL_QUEUE_NAME
import ch.loewenfels.issuetrackingsync.scheduling.BrokerFacade
import ch.loewenfels.issuetrackingsync.scheduling.QueueStatistics
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class StatisticsController(private val brokerFacade: BrokerFacade) {
    @GetMapping("/statistics")
    fun statistics(): QueueStatistics? =
        brokerFacade.getStatistics(INTERNAL_QUEUE_NAME)
}
